ui-core/src/components/Select.tsx

```diff
@@ -2,88 +2,72 @@ import React, { useState } from 'react';
 import FieldWrapper, { FieldWrapperProps } from './inputs/FieldWrapper';
 import cx from 'classnames';
 
-export type SelectOption = { value: string; label: string };
-
-/**
- * Type for the props of the Select component.
- *
- * We use Omit to create a new type based on React.InputHTMLAttributes<HTMLSelectElement>,
- * but without the 'value' property. This is because we want to enforce that the 'value'
- * prop for our Select component is always a string. In React, the 'value' prop for a
- * select element can be a string, a number, or an array of strings (for multiple selects),
- * but for our Select component, we want to ensure that it's always a string.
- */
-export type SelectProps = Omit<React.InputHTMLAttributes<HTMLSelectElement>, 'value'> &
+export type SelectProps<T> = Omit<
+  React.InputHTMLAttributes<HTMLSelectElement>,
+  'value' | 'onChange'
+> &
   Omit<FieldWrapperProps, 'children'> & {
-    value?: string;
-    placeholder?: string;
-    options: SelectOption[];
+    options: Array<T>;
+    value?: T;
```

Comment:
If value can be undefined, it should be handle in T
```suggestion
    value: T;
```